### PR TITLE
create qa view for overlapping SF enrollments

### PR DESCRIPTION
**Pull request by anthonygwalters**
# Pull Request

## Summary & Motivation

[//]: # "When merged, this pull request will..."

## Self-review

- [ ] Update **due date**, **assignee**, and **priority** on our
      [TEAMster Asana Project](https://app.asana.com/0/1205971774138578/1205971926225838)

  _If this is a same-day request, please flag that in Slack_

- [ ] <kbd>Format</kbd> has been run on all modified files

- [ ] Ensure you are using the `union_dataset_join_clause()` macro for queries that employ any
      models from these datasets:
  - deanslist
  - edplan
  - iready
  - pearson
  - powerschool
  - renlearn
  - titan

## Troubleshooting

- [SqlFluff Rules Reference](https://docs.sqlfluff.com/en/stable/rules.html)
- [Trunk](https://teamschools.github.io/teamster/CONTRIBUTING/#trunk)
- [dbt](https://teamschools.github.io/teamster/CONTRIBUTING/#dbt-cloud_1)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207270856009428

### CODE REVIEW

File: src/dbt/kipptaf/models/kippadb/qa/qa_kippadb__duplicate_enrollments.sql [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                )
+                or e2.actual_end_date between e1.start_date and coalesce(
+                    e1.actual_end_date, date(2024, 06, 30)
+                )
+            )
+            and e2.status != 'Did Not Enroll'
+        inner join {{ ref("int_kippadb__roster") }} as r on e1.student = r.contact_id
+        where
+            e1.pursuing_degree_type
+            in ("Associate's (2 year)", "Associate's (2 year)", 'Certificate')
```
Comment (cbini) [inline src/dbt/kipptaf/models/kippadb/qa/qa_kippadb__duplicate_enrollments.sql:47]: should this be bachelors?

File: src/dbt/kipptaf/models/kippadb/qa/qa_kippadb__duplicate_enrollments.sql [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            and e1.id != e2.id
+            and (
+                e2.start_date between e1.start_date and coalesce(
+                    e1.actual_end_date, date(2024, 06, 30)
+                )
+                or e2.actual_end_date between e1.start_date and coalesce(
+                    e1.actual_end_date, date(2024, 06, 30)
+                )
+            )
+            and e2.status != 'Did Not Enroll'
```
Comment (cbini) [inline src/dbt/kipptaf/models/kippadb/qa/qa_kippadb__duplicate_enrollments.sql:43]: could this be done via lag? could you explain in words what you're looking to do here?

File: src/dbt/kipptaf/models/kippadb/qa/qa_kippadb__duplicate_enrollments.sql [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            ) as enrollment_id_1_lag,
+        from {{ ref("stg_kippadb__enrollment") }} as e1
+        inner join
+            {{ ref("stg_kippadb__enrollment") }} as e2
+            on e1.student = e2.student
+            and e1.pursuing_degree_type = e2.pursuing_degree_type
+            and e1.id != e2.id
+            and (
+                e2.start_date between e1.start_date and coalesce(
+                    e1.actual_end_date, date(2024, 06, 30)
```
Comment (cbini) [inline src/dbt/kipptaf/models/kippadb/qa/qa_kippadb__duplicate_enrollments.sql:37]: hard-coded year. should this use a variable?

**Review by cbini — CHANGES_REQUESTED** [review]

File: src/dbt/kipptaf/models/kippadb/qa/qa_kippadb__duplicate_enrollments.sql (hunk 2 above)
Comment (anthonygwalters) [reply src/dbt/kipptaf/models/kippadb/qa/qa_kippadb__duplicate_enrollments.sql:43]: not sure re: lag, but the purpose is just to catch all forms of overlap on enrollments of the same degree type

File: src/dbt/kipptaf/models/kippadb/qa/qa_kippadb__duplicate_enrollments.sql (hunk 3 above)
Comment (anthonygwalters) [reply src/dbt/kipptaf/models/kippadb/qa/qa_kippadb__duplicate_enrollments.sql:37]: fixed - will be on next commit

File: src/dbt/kipptaf/models/kippadb/qa/qa_kippadb__duplicate_enrollments.sql (hunk 1 above)
Comment (anthonygwalters) [reply src/dbt/kipptaf/models/kippadb/qa/qa_kippadb__duplicate_enrollments.sql:47]: yes, yes it should - fixed on next commit

**Review by cbini — APPROVED** [review]

**Review by cbini — APPROVED** [review]

**Pull request merged**